onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -768,45 +770,244 @@
 
   if (num_events > 0) {
     LOGS(*logger_, VERBOSE) << "profile_events: " << profile_events << " num_events: " << num_events;
-  }
 
-  for (size_t event_idx = 0; event_idx < num_events; event_idx++) {
-    ORT_RETURN_IF_ERROR(ExtractProfilingEvent(*(profile_events + event_idx)));
-    ORT_RETURN_IF_ERROR(ExtractProfilingSubEvents(*(profile_events + event_idx)));
+    bool backendSupportsExtendedEventData = false;
+    Qnn_ErrorHandle_t resultPropertyHasCapability =
+        qnn_interface_.propertyHasCapability(QNN_PROPERTY_PROFILE_SUPPORTS_EXTENDED_EVENT);
+    uint16_t errorCodePropertyHasCapability = static_cast<uint16_t>(resultPropertyHasCapability & 0xFFFF);
+    if (errorCodePropertyHasCapability == QNN_PROFILE_NO_ERROR) {
+      LOGS(*logger_, VERBOSE) << "The QNN backend supports extended event data.";
+      backendSupportsExtendedEventData = true;
+    } else {
+      LOGS(*logger_, VERBOSE) << "The QNN backend does not support extended event data.";
+    }
+
+    // Write to CSV in append mode
+    const char* profilingCsvFilename = "qnn-profiling-data.csv";
+    std::ifstream infile(profilingCsvFilename);
+    bool exists = infile.good();
+    infile.close();
+
+    std::ofstream outfile(profilingCsvFilename, std::ios_base::app);
```

Comment:
>std::ofstream outfile(profilingCsvFilename, std::ios_base::app); [](start = 4, length = 64)

Should we always create new one? With the append mode, no way to know whether it's a new model.
Also add time stamp to the file name to identify new runs.

Reply:
This works as well however I am not sure it is "better" - just different with different pros/cons. We definitely would want to handle cases where multiple models run in succession, and you can gather output from multiple models. It's fairly normal for logs to append and you can find the file easily with a static name. Your approach is ok but then you juggle multiple files and automating is a little trigger w/o consistent names. Not 100% tied to either approach but I personally like the append mode as-is better